### PR TITLE
Promote CronJobControllerV2 flag to GA

### LINKED ISSUE #27394: CronJob concept implies that CronJob can schedule one-shot tasks for a specific future date
**This is a Bug Report**

**Problem:**
The [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) concept page mentions:
> CronJobs can also schedule individual tasks for a specific time, such as scheduling a Job for when your cluster is likely to be idle.

This sounds like [`at`](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/at.html), which can schedule one-shot tasks for a specific future date / datetime; however, that page strongly implies that CronJob does not in fact provide “run exactly once and never repeat” semantics.

Technically you could define a CronJob and a ServiceAccount and authnz rules that let the CronJob delete itself after it has successfully run once. I don't think we want to mention that bodge. I'd rather see an addon that gives you a OneShotScheduledJob CRD or similar.

**Proposed Solution:**

Remove or reword the text “CronJobs can also schedule individual tasks for a specific time, such as scheduling a Job for when your cluster is likely to be idle.”

**Page to Update:**

https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/

**Additional Information:**
/kind bug
/sig apps

### PR DESCRIPTION
**Pull request by soltysh**
This goes in pair with https://github.com/kubernetes/kubernetes/pull/102529

Fixes #27394

/assign @alaypatel07

### CODE REVIEW

**Comment by PI-Victor** [comment]
i don't see this graduation being tracked for 1.22, @soltysh can you confirm that this is tracked currently for 1.22?

also hold for the pr to land in case it is.
/hold

**Comment by PI-Victor** [comment]
we've been talking about this in the RT team and it seems that it might need an exception to be tracked for 1.22
cc @JamesLaverack @annajung

**Comment by soltysh** [comment]
> we've been talking about this in the RT team and it seems that it might need an exception to be tracked for 1.22
> cc @JamesLaverack @annajung

@PI-Victor good catch, it slipped my attention, I filled in the exception request in https://groups.google.com/g/kubernetes-sig-release/c/LejSpoeCiWg

**Review by alaypatel07 — COMMENTED** [review]
/lgtm

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (PI-Victor) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:98]: since this is now graduating, it needs to be moved below in the table [Feature gates for graduated or deprecated features](https://github.com/kubernetes/website/blob/a2eac5b39e7ae9adb961879552a9d76e7595f994/content/en/docs/reference/command-line-tools-reference/feature-gates.md#feature-gates-for-graduated-or-deprecated-features)

**Review by PI-Victor — CHANGES_REQUESTED** [review]

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (PI-Victor) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:98]: sorry, just to be more succinct, all `CronJobControllerV2` entries from here now will go in that table.

**Comment by PI-Victor** [comment]
also the description for the controller on the same page says: 

> CronJobControllerV2: Use an alternative implementation of the {{< glossary_tooltip text="CronJob" term_id="cronjob" >}} controller. Otherwise, version 1 of the same controller is selected. The version 2 controller provides experimental performance improvements.

Is it safe to say that `The version 2 controller provides experimental performance improvements.` are no longer experimental, since it's now graduating? if that's still the case, we can leave it as it is.

**Comment by soltysh** [comment]
> Is it safe to say that `The version 2 controller provides experimental performance improvements.` are no longer experimental, since it's now graduating? if that's still the case, we can leave it as it is.

Yeah, good point updated. 

I'm also including a fix to #27394.

**Comment by soltysh** [comment]
@PI-Victor should be all good now

**Comment by PI-Victor** [comment]
appreciate it, thank you!
/lgtm

**Comment by PI-Victor** [comment]
since this is now marked as tracked and the enhancement has green light, i'll go ahead and remove the hold
/hold cancel

**Pull request merged**
